packages/material-ui/test/typescript/components.spec.tsx

```diff
@@ -949,6 +949,17 @@ const SelectTest = () => (
   </Select>
 );
 
+const MultipleSelectTest = () => (
+  <Select multiple input={<Input />} value={[10]} onChange={e => log(e.currentTarget.value)}>
```

Comment:
This is the issue I was referring to in https://github.com/mui-org/material-ui/pull/16090#issuecomment-500329906.

TypeScript will infer `number[]` but the very first item already has a string. The type information is now incorrect without any indication. By requiring type casting we at least make it explicit that we have unsound types. For sound types runtime checking is an option.